src/qml/NavigatingDrawer.qml

```diff
@@ -0,0 +1,125 @@
+import QtQuick 2.14
+import QtQuick.Controls 2.14
+
+import org.qfield 1.0
+import Theme 1.0
+
+Drawer {
+  id: controller
+  width: parent.width
+  height: mainContent.height
+  Overlay.modal: null
+  modal: false
+  edge: Qt.BottomEdge
+  dragMargin: 0
+  closePolicy: Popup.NoAutoClose
+
+  property real realtimeHeight: controller.height * controller.position
+  property bool closeRequested: true
+  property bool openRequested: false
+
+  // NavigationInformationView
+  property bool navigationInformationViewEnabled: navigation.isActive
+
+  // PositioningInformationView
+  property Navigation navigation
+  property double antennaHeight
+  property bool positioningInformationViewEnabled: false
+
+  // PositioningPreciseView
+  property alias positioningPreciseView: positioningPv
+  property PositioningSettings positioningSettings
+  property Positioning positionSource
+  property bool positioningPreciseEnabled: false
+  property real positioningPreciseViewHeight
+
+  function resetHeight() {
+    let newHeight = 0
+    newHeight += (navigationInformationView.height + 8)
+    newHeight += (positioningInformationView.height + 8)
+    newHeight += (positioningPv.height + 8)
+    newHeight += 16
+    controller.height = newHeight
+  }
+
+  function updateDrawerHeight(visibleView, viewHeight){
+    if (!visibleView) {
+      controller.height -= (viewHeight + 8)
+      controller.open()
+    } else if (openRequested) {
+      controller.height += (viewHeight + 8)
+      controller.open()
+    }
+  }
+
+  onNavigationInformationViewEnabledChanged: updateDrawerHeight (navigationInformationViewEnabled, navigationInformationView.contentHeight)
```

Comment:
To be consistent, let's avoid spaces between the function name and the (

```suggestion
  onNavigationInformationViewEnabledChanged: updateDrawerHeight(navigationInformationViewEnabled, navigationInformationView.contentHeight)
```

Reply:
sure